packages/url_launcher/url_launcher_web/test/integration_test/url_launcher_web_test.dart

```diff
@@ -2,35 +2,39 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// @dart = 2.9
 import 'dart:html' as html;
-import 'dart:js_util';
-import 'package:flutter/widgets.dart';
 import 'package:flutter_test/flutter_test.dart';
-import 'package:url_launcher_platform_interface/link.dart';
+import 'package:mockito/annotations.dart';
 import 'package:url_launcher_web/url_launcher_web.dart';
-import 'package:url_launcher_web/src/link.dart';
 import 'package:mockito/mockito.dart';
 import 'package:integration_test/integration_test.dart';
 
-class _MockWindow extends Mock implements html.Window {}
-
-class _MockNavigator extends Mock implements html.Navigator {}
+import 'url_launcher_web_test.mocks.dart';
 
+@GenerateMocks([], customMocks: [
+  MockSpec<html.Window>(returnNullOnMissingStub: true),
+  MockSpec<html.Navigator>(returnNullOnMissingStub: true)
+])
 void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
 
   group('UrlLauncherPlugin', () {
-    _MockWindow mockWindow;
-    _MockNavigator mockNavigator;
+    late MockWindow mockWindow;
+    late MockNavigator mockNavigator;
 
-    UrlLauncherPlugin plugin;
+    late UrlLauncherPlugin plugin;
 
     setUp(() {
-      mockWindow = _MockWindow();
-      mockNavigator = _MockNavigator();
+      mockWindow = MockWindow();
+      mockNavigator = MockNavigator();
       when(mockWindow.navigator).thenReturn(mockNavigator);
 
+      // Simulate that window.open does something.
+      when(mockWindow.open(any, any)).thenReturn(MockWindow());
+
+      when(mockNavigator.vendor).thenReturn('Google LLC');
+      when(mockNavigator.appVersion).thenReturn('Mock version!');
```

Comment:
According to dart:html these can't be null, so we must mock them by default. These are used pretty much for everything the plugin does!